### PR TITLE
Cursor: Implement overlap checks for the cursor

### PR DESCRIPTION
**Pull request by TheAirMike**
The cursor will now check whether the mesh will overlap with existing pieces and prevent the user from placing it.

New UI for controls and overlap status.

### CODE REVIEW

**Review by cooltrain7 — CHANGES_REQUESTED** [review]
Everything to do with the overlap looks good. Only nitpicks are with the UI.

I would:
- Shrink the font back to default 16/18px
- Add some padding to the vbox from the side/top, 5px should do
- Give the labels real names

Some other food for thought is what we want to do going forward with all these debug prints. If you want to remove/keep certain ones around.

File: editor/script/cursor.gd [hunk 1]
```diff
@@ -2,11 +2,54 @@ extends Node3D
...
+		overlap_area.set_collision_mask_value(2, true)
+		
+		# Connect colision signals to the appropriate functions
+		overlap_area.connect("area_entered", _on_overlap_begin)
+		overlap_area.connect("area_exited", _on_overlap_stop)
+
 @export var cursor_mesh: Buildable
 @export var old_mesh: Buildable
 
+var unplacable: bool = false :
```
Comment (cooltrain7) [inline editor/script/cursor.gd]: Personally I think its better to have positive variable names over negative. `is_placeable `over a `is_not_placeable`. Helps with readability when you have to invert `!unplaceable` in your head.

File: editor/script/cursor.gd [hunk 2]
```diff
@@ -15,10 +58,11 @@ func _input(event):
 			
 	if event is InputEventMouseButton:
 		if event.pressed and event.button_index == 1:
-			if cursor_mesh != null:
-				var instance = cursor_mesh.scene.instantiate()
-				(instance as Node3D).position = position
-				get_parent().add_child(instance)
+			if !unplacable:
```
Comment (cooltrain7) [inline editor/script/cursor.gd]: Could compress this down if we don't need a specific else for either of these

`if is_placeable and cursor_mesh != null`

**Review by cooltrain7 — DISMISSED** [review]
Happy with the UI, did some more nitpicking that I missed the first time. Approve regardless of outcome.

File: editor/script/cursor.gd (hunk 1 above)
Comment (TheAirMike) [reply editor/script/cursor.gd]: I agree, updated👍

File: editor/script/cursor.gd (hunk 2 above)
Comment (TheAirMike) [reply editor/script/cursor.gd]: Was a leftover from the `else` clause which I removed, updated 👍

**Review by cooltrain7 — APPROVED** [review]
All good.

**Pull request merged**